src/main/java/net/minestom/server/utils/incubator/Pos.java

```diff
@@ -0,0 +1,70 @@
+package net.minestom.server.utils.incubator;
+
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+public class Pos implements Vec {
+    public static final Pos ZERO = new Pos();
+
+    private final double x, y, z;
+    private final float yaw, pitch;
+
+    public Pos(double x, double y, double z, float yaw, float pitch) {
+        this.x = x;
+        this.y = y;
+        this.z = z;
+        this.yaw = yaw;
+        this.pitch = pitch;
+    }
+
+    public Pos(double x, double y, double z) {
+        this(x, y, z, 0, 0);
+    }
+
+    public Pos() {
+        this(0, 0, 0, 0, 0);
+    }
+
+    @Contract(pure = true)
+    public @NotNull Pos withYaw(float yaw) {
```

Comment:
Should probably have checks to see if the values are actually changing, otherwise we can just return the current pos.